src/ports/postgres/modules/utilities/minibatch_preprocessing.py_in

```diff
@@ -0,0 +1,559 @@
+                       "cannot be null")
+
+    def get_query_for_standardizing(self):
+        query="""
+        SELECT
+        {dep_var_array_str} as {dep_colname},
+        {schema_madlib}.utils_normalize_data
+        (
+            {indep_var_array_str},'{x_mean_str}'::double precision[],
+            '{x_std_dev_str}'::double precision[]
+        ) as {ind_colname}
+        FROM {source_table}
+        """.format(
+            source_table = self.source_table,
+            schema_madlib = self.schema_madlib,
+            dep_var_array_str = self.dep_var_array_str,
+            indep_var_array_str = self.indep_var_array_str,
+            dep_colname = MINIBATCH_OUTPUT_DEPENDENT_COLNAME,
+            ind_colname = MINIBATCH_OUTPUT_INDEPENDENT_COLNAME,
+            x_mean_str = self.x_mean_str,
+            x_std_dev_str = self.x_std_dev_str)
+        return query
+
+    def create_output_standardization_table(self):
+        query = """
+        CREATE TABLE {output_standardization_table} AS
+        select {grouping_cols}::TEXT AS grouping_cols,
+        '{x_mean_str}'::double precision[] AS mean,
+        '{x_std_dev_str}'::double precision[] AS std
+        """.format(
+        output_standardization_table = self.output_standardization_table,
+        grouping_cols = self.grouping_cols,
+        x_mean_str = self.x_mean_str,
+        x_std_dev_str = self.x_std_dev_str)
+        plpy.execute(query)
+
+class MiniBatchSummarizer:
+    @staticmethod
+    def create_output_summary_table(source_table, output_table,
+                                    dep_var_array_str, indep_var_array_str,
+                                    buffer_size, class_values, num_rows_processed,
+                                    num_missing_rows_skipped, output_summary_table):
+        query = """
+            CREATE TABLE {output_summary_table} AS
+            SELECT '{source_table}'::TEXT AS source_table,
+            '{output_table}'::TEXT AS output_table,
+            '{dependent_varname}'::TEXT AS dependent_varname,
+            '{independent_varname}'::TEXT AS independent_varname,
+            {buffer_size} AS buffer_size,
+            {class_values} AS class_values,
+            {num_rows_processed} AS num_rows_processed,
+            {num_missing_rows_skipped} AS num_missing_rows_skipped,
+            {grouping_cols}::TEXT AS grouping_cols
+        """.format(output_summary_table = output_summary_table,
+                   source_table = source_table,
+                   output_table = output_table,
+                   dependent_varname = dep_var_array_str,
+                   independent_varname = indep_var_array_str,
+                   buffer_size = buffer_size,
+                   class_values = class_values,
+                   num_rows_processed = num_rows_processed,
+                   num_missing_rows_skipped = num_missing_rows_skipped,
+                   grouping_cols = "NULL")
+        plpy.execute(query)
```

Comment:
I tried running the module, without setting any value for `buffer_size`. In the output table, each buffer had exactly one row (due to the work in progress formula used in `MiniBatchBufferSizeCalculator`), but the summary table reported `buffer_size = 38.0`.
`buffer_size` in the output table and summary table are in sync when I specify a value for `buffer_size`, but there seems to be an issue when we don't specify the `buffer_size`.

Reply:
good catch. The problem is that we assumed that math.ceil returns an int but it returns a float and hence the bug. Our unit tests could not catch this because comparing an int value for a float variable is valid. We need to assert that the instance type of the variable is int. 
We will improve the unit test as well as fix the bug.